internal/util/observability/observability.go

```diff
@@ -19,34 +19,82 @@ package observability
 
 import (
 	"context"
+	"errors"
 	"time"
 
 	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
 	otelsdkresource "go.opentelemetry.io/otel/sdk/resource"
 	otelsdktrace "go.opentelemetry.io/otel/sdk/trace"
 	otelsemconv "go.opentelemetry.io/otel/semconv/v1.21.0"
-
-	"github.com/FerretDB/FerretDB/internal/util/must"
+	"go.uber.org/zap"
 )
 
-// SetupOtel sets up OpenTelemetry exporter with fixed values.
-func SetupOtel(service string) func(context.Context) error {
-	exporter := must.NotFail(otlptracehttp.New(
+// OtelConfig is the configuration for OpenTelemetry.
+type OtelConfig struct {
+	Service  string
+	Version  string
+	Endpoint string
+}
+
+// Otel represents the OpenTelemetry system.
+type Otel struct {
+	l  *zap.Logger
+	tp *otelsdktrace.TracerProvider
+}
+
+// NewOtel sets up OTLP exporter and tracer provider.
+func NewOtel(config *OtelConfig, l *zap.Logger) (*Otel, error) {
```

Comment:
We generally don't pass options structure and separate arguments at the same time. We could do `NewOtelOpts` as in other places